### PR TITLE
refine trie search benchmark

### LINKED ISSUE #13: create a dir for all kinds of bench/profiling report/charts

### PR DESCRIPTION
**Pull request by wenbobuaa**
# Description

### refine slimtrie search benchmark 
   1. rename: trie/util.go -> trie/test_util.go
   2. add func: makeTrieBenchFunc, makeMapBenchFunc, ... 

Fixes #13 

## Type of change

Please delete options that are not relevant.

- [x] Refactoring
- [x] Document changes
- [x] Test changes

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -81,6 +81,10 @@
   name = "gopkg.in/yaml.v2"
   version = "2.2.2"
 
+[[constraint]]
+  branch = "master"
+  name = "github.com/modood/table"
```
Comment (drmingdrmer) [inline Gopkg.toml]: 只更新了toml文件?lock文件没更新? vendor里也没更新吗?

File: trie/test_utils.go [hunk 2]
```diff
@@ -0,0 +1,375 @@
+package trie
```
Comment (drmingdrmer) [inline trie/test_utils.go]: 有些东西...能不放xx_util里就不放吧...很容易最后啥都在util里...

File: tools/app/trie_search_cost.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package main
```
Comment (drmingdrmer) [inline tools/app/trie_search_cost.go]: 这个程序有注释或文档吗? 大致描述下做啥, 输出啥的

File: Gopkg.toml (hunk 1 above)
Comment (wenbobuaa) [reply Gopkg.toml]: `dep ensure` 一直网络问题，新的提交点里好了

File: tools/app/trie_search_cost.go (hunk 3 above)
Comment (wenbobuaa) [reply tools/app/trie_search_cost.go]: 在 `doc/trie/benchmark_readme.md` 有介绍。我再加1点注释

File: trie/test_utils.go (hunk 2 above)
Comment (wenbobuaa) [reply trie/test_utils.go]: 嗯嗯 拆一下

Comment (wenbobuaa) [reply trie/test_utils.go]: 看了一遍好像没什么能拆出去的，都得放在test_util..

File: Gopkg.toml (hunk 1 above)
Comment (drmingdrmer) [reply Gopkg.toml]: 我都是登到香港的vps上dep下来,push,然后回本地继续调...

File: tools/app/trie_search_cost.go (hunk 3 above)
Comment (drmingdrmer) [reply tools/app/trie_search_cost.go]: 感觉那个文档里直接说的是怎么做, 开头可以1,2句提下, 为啥做, 做啥, 做完之后生成啥东西.

然后所有"怎么做"的步骤里, 哪些步骤是为了输出啥东西也不太清楚, 最后这些步骤里哪些必须做哪些是可选的也不清楚.

大概这样: 

```
1-生成bench report, 需要以下步骤:
 1. install
 2. run it.
 3. open xx.md
 4. xx.md里包括xx,yy信息

2-生成性能火焰图, 需要以下步骤:
 1. install
 2. bench it
 3. open xx.svg
 4. ...

...
```

File: docs/trie/benchmark_readme.md [hunk 4]
```diff
@@ -1,45 +1,55 @@
...
+It can be considered that the less cost the better performance in slimtrie, map and Btree.
 
+There is a [benchmark comparision](benchmark_result.md) report.
 Benchmark uses the Go language built-in `map` and the [BTree implementation for Go](https://github.com/google/btree).
 
+Benchmark and optimization steps are as follows:
+
+## Do Benchmark
+
 1. `go test -cpuprofile cpu.profiling -benchmem -bench=. -run=none`
```
Comment (drmingdrmer) [inline docs/trie/benchmark_readme.md]: 这里为啥要加`-cpuprofile`啊?

File: docs/trie/benchmark_readme.md [hunk 5]
```diff
@@ -1,45 +1,55 @@
...
-3. `go run trie_search_cost.go`
+4. `go run trie_search_cost.go` (optional)
 
-> this command in `slim/tools/app/` runs key search benchmark and output a chart result
-> for a better view.
+> SlimTrie offers a tool in `slim/tools/app/` to run key search benchmark and output a chart result in
+> string. It show a better view than `go test -bench`.
 
 
 ## Use Flamegraph
```
Comment (drmingdrmer) [inline docs/trie/benchmark_readme.md]: 恩...感觉这个readme是2部分, 一个是benchmark, 一个是profiling. 这里标题写: `profiling with flamegraph`感觉好点.

光说Use flamegraph感觉不明白为啥要用它.

File: docs/trie/benchmark_readme.md (hunk 4 above)
Comment (wenbobuaa) [reply docs/trie/benchmark_readme.md]: 这里是不用加 去掉

**Pull request merged**